### PR TITLE
Setup Pipeline and Gradle Configuration for Publishing the library

### PR DESCRIPTION
**Pull request by mayorJAY**

### CODE REVIEW

**Review by unicodeveloper — APPROVED** [review]

File: .github/workflows/maven-publish.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      - name: Validate Gradle wrapper
+        uses: gradle/wrapper-validation-action@ccb4328a959376b642e027874838f60f8e596de3
+
+      - name: Publish package
+        uses: gradle/gradle-build-action@749f47bda3e44aa060e82d7b3ef7e40d953bd629
+        with:
+          arguments: publish
+        env:
+          MAVEN_USERNAME: ${{ secrets.OSSRH_USERNAME }}
+          MAVEN_PASSWORD: ${{ secrets.OSSRH_TOKEN }}
```
Comment (unicodeveloper) [inline .github/workflows/maven-publish.yml:37]: I think this OSSRH_TOKEN should be OSSRH_PASSWORD @mayorJAY

File: build.gradle.kts [hunk 2]
```diff
@@ -80,13 +79,13 @@ publishing {
             }
             pom {
                 name.set("Novu Kotlin")
-                description.set("A kotlin SDK for Novu")
-                url.set("https://github.com/Crashiv/novu-kotlin")
+                packaging = "jar"
+                description.set("Kotlin SDK for Novu - The open-source notification infrastructure for engineers")
```
Comment (unicodeveloper) [inline build.gradle.kts:83]: engineers => developers

File: .github/workflows/maven-publish.yml (hunk 1 above)
Comment (mayorJAY) [reply .github/workflows/maven-publish.yml:37]: Yes, it is

File: build.gradle.kts (hunk 2 above)
Comment (mayorJAY) [reply build.gradle.kts:83]:
```suggestion
                description.set("Kotlin SDK for Novu - The open-source notification infrastructure for developers")
```

File: .github/workflows/maven-publish.yml (hunk 1 above)
Comment (mayorJAY) [reply .github/workflows/maven-publish.yml:37]:
```suggestion
          MAVEN_PASSWORD: ${{ secrets.OSSRH_PASSWORD }}
```

**Pull request merged**